kratos-admin-ui/src/components/traits/edit-traits.tsx

```diff
@@ -128,15 +160,15 @@ export function EditTraits(props: EditTraitsProps) {
                 valueObject = {
                     state: identity.state!,
                     traits: await fillTraits(identity, newSchemaFields),
-                    publicMetadata: identity.metadata_public,
-                    adminMedataData: identity.metadata_admin
+                    publicMetadata: await fillMetadata(identity, newSchemaFields, "metadata_public"),
+                    adminMetadata: await fillMetadata(identity, newSchemaFields, "metadata_admin")
                 }
             } else if (props.schema && props.modi === "new") {
                 newSchemaFields = SchemaService.getSchemaFields(props.schema)
                 valueObject = {
                     state: "active",
                     traits: {},
-                    adminMedataData: {},
+                    adminMetadata: {},
```

Comment:
I just checked the code, i wasnt able to create a new identity with metadata - the values just get dropped. I think we have to include the code from the lines 163 and 164.

Reply:
Interesting, I am not able to reproduce that - on our Kratos v1.0.0 installation, creating and updating identities with public or admin metadata works well. Could you perhaps try it with this schema? https://gist.github.com/toonalbers/d4b4fcf2472404066de47061ca37cc3c